### ISSUE TITLE
[BUG] Reopen JS template literal close indent issue #475

### ISSUE THREAD

**Issue by jpradelle** [yeonjuan/html-eslint#586]
I'm reopening issue #475 
There is conflict that can't be resolved since version `0.52.0`

I'm stuck with stuff explained in my last comment https://github.com/yeonjuan/html-eslint/issues/475#issuecomment-3755467628

There is no way to allow this syntax because it's reported as an issue
```js
      ${when(
          true,
          () => html`
            <b>ok</b>
          `)}
```
[playground link](https://html-eslint.org/playground#eyJjb2RlIjoiaHRtbGBcbiAgPGRpdj5cbiAgICAke3doZW4oXG4gICAgICAgICAgdHJ1ZSxcbiAgICAgICAgICAoKSA9PiBodG1sYFxuICAgICAgICAgICAgPG9rPjwvb2s+XG4gICAgICAgICAgYCl9XG4gIDwvZGl2PlxuYDsiLCJjb25maWciOnsicnVsZXMiOnsiQGh0bWwtZXNsaW50L2luZGVudCI6WyJlcnJvciIsMix7IkF0dHJpYnV0ZSI6Mn1dfX0sImxhbmd1YWdlIjoiamF2YXNjcmlwdCJ9)

html-eslint ask one of those syntaxes
```js
      ${when(
          true,
          () => html`
            <b>ok</b>
      `)} // <-- I think this one is an issue

      ${when(
          true,
          () => html`
            <b>ok</b>
          `)
      }
```

I think you should check indent of closing bracket only if this is the first item of the line.

`@stylistic/js/indent` checks it too, agrees with the rule on that point :
<img width="691" height="233" alt="Image" src="https://github.com/user-attachments/assets/8ad66e10-39b1-4cc2-b122-15399d15ab5c" />

But `@stylistic/js/indent` allows this one where `html-eslint` doesn't:
<img width="659" height="186" alt="Image" src="https://github.com/user-attachments/assets/92375e4d-e077-47f2-8b25-8b945cfc07ea" />

My initial intention for the request was too check indent of closing backtick of template literal https://github.com/yeonjuan/html-eslint/issues/462
After you said you made it on `${` and `}` for backticks we should delegate to an other lib https://github.com/yeonjuan/html-eslint/issues/475#issuecomment-3752257048
`@stylistic/js/indent` rule seems to do it, maybe update made on version `0.52.0` is a misunderstanding and can be revert as `@stylistic/js/indent` do the job on `${` and `}` ?

**Comment by yeonjuan** [comment]
Hi @jpradelle Sorry, I missed your comment earlier. Thanks for reopening the issue.

> I think you should check indent of closing bracket only if this is the first item of the line.

I agree. It seems the current behavior is a bug. Let me fix it

**Comment by yeonjuan** [comment]
@jpradelle 
Hmm... after looking into this, it turns out this is the intended behavior. It’s not checking for backticks or `}` specifically, but rather the indentation of the child content.

```js
html`
  <div>
    ${when(
          true,
          () => html`
            <ok></ok>
          `)} // <-- it's child of div
  </div>
`;
```

Are you perhaps trying to align the indentation levels of the parentheses(`(`, `)`) for the `when` function?
Would it work if there were an option to completely ignore indentation when a template literal expression is used as child content?

**Comment by jpradelle** [comment]
I can agree it's child of div, but other parts are child of div too and are not reported as issue.

Fully exploded indent would look like and comply to the linter
```js
html`
  <div>
    ${
      when(
          true,
          () => html`
            <ok></ok>
          `
      )
    }
  </div>
`;
```
But to compress a the syntax I would like to remove some line return. Like I think it's a common way to do with functions, and that's what I'm looking for.
```js
this.foo(
    this.bar,
    this.baz(
        this.baaz()));
// or
this.foo(
    this.bar,
    this.baz(
        this.baaz()
    ));

// Instead of
this.foo(
    this.bar,
    this.baz(
        this.baaz()
    )
);
```

And if I try to do this logic with JS in template this breaks.

Here is another sample of issue

<img width="773" height="259" alt="Image" src="https://github.com/user-attachments/assets/81622590-8cff-4617-a74f-226e8be25596" />

If I'm not wrong all of those syntaxes were allowed before version 0.52.0. 

It's not easy to find the good way for mixing languages

> Would it work if there were an option to completely ignore indentation when a template literal expression is used as child content?

In lit, I think this is a normal thing to have multiple templates levels (In my case at least I have a lot). And your linter works very well with it, so I don't think adding an option would not be a good idea, we would not lint a very large part of the code instead, and today this is working well, unless this issue.

**Comment by yeonjuan** [comment]
@jpradelle 
> I think you should check indent of closing bracket only if this is the first item of the line.

I see. I understand. It seems that making improvements using the method you mentioned is the best approach. 

After the modification, if the code looks like the example below, "bar" must be modified so that no indentation errors occur. 

https://html-eslint.org/playground#eyJjb2RlIjoiaHRtbGBcbiAgICA8ZGl2PlxuICAgICAgICAke1xuXHRcdFxuICBcdFx0XHRhID8gXCJmb29cIjpcbiAgICAgICAgXCJiYXJcIn1cbiAgICA8L2Rpdj5cbmBcbiIsImNvbmZpZyI6eyJydWxlcyI6eyJAaHRtbC1lc2xpbnQvaW5kZW50IjoiZXJyb3IifX0sImxhbmd1YWdlIjoiamF2YXNjcmlwdCJ9
```html
html`
    <div>
        ${
		
  			a ? "foo":
        "bar"}
    </div>
`
```

**Comment by yeonjuan** [comment]
@jpradelle Hi, I've released `v0.60.0-alpha.1` which includes a fix for this. If you have some time, could you please check if it works as expected in this version?

```
// npm
npm install -D @html-eslint/eslint-plugin@0.60.0-alpha.1
// yarn
yarn add -D @html-eslint/eslint-plugin@0.60.0-alpha.1
```

**Comment by jpradelle** [comment]
Hi, thank you for the update.

About that I don't agree, and `@stylistic/js` eslint plugin doesn't agree too.
> [@jpradelle](https://github.com/jpradelle)
> 
> > I think you should check indent of closing bracket only if this is the first item of the line.
> 
> I see. I understand. It seems that making improvements using the method you mentioned is the best approach.
> 
> After the modification, if the code looks like the example below, "bar" must be modified so that no indentation errors occur.
> 
> https://html-eslint.org/playground#eyJjb2RlIjoiaHRtbGBcbiAgICA8ZGl2PlxuICAgICAgICAke1xuXHRcdFxuICBcdFx0XHRhID8gXCJmb29cIjpcbiAgICAgICAgXCJiYXJcIn1cbiAgICA8L2Rpdj5cbmBcbiIsImNvbmZpZyI6eyJydWxlcyI6eyJAaHRtbC1lc2xpbnQvaW5kZW50IjoiZXJyb3IifX0sImxhbmd1YWdlIjoiamF2YXNjcmlwdCJ9
> ```js
> html`
>     <div>
>         ${
> 		
>   			a ? "foo":
>         "bar"}
>     </div>
> `
> ```

With version `0.60.0-alpha.1` This is good with ternary: this is allowed
```js
html`
  <div>
    ${a
      ? 'foo'
      : 'bar'}
  </div>
`;
```

I still have the issue with this pattern not allowed
```js
html`
  ${when(
      true,
      () => html`
        <b>ok</b>
      `)}
`;
```

I think this sample shows a bit more the issue, this leads to conflicts with other javascript linter:

<img width="738" height="292" alt="Image" src="https://github.com/user-attachments/assets/92306d98-5b75-4f6a-87fa-35edf69d9c31" />

And I agree with `@stylistic/js` here

I really think `@html-eslint/indent` should check the ident of the closing brace only if this is the first item of the line. If there is some content before the closing brace, this is in the scope of the javascript linter to check the indent.

If I transpose this javascript structure
```js
html`
  ${when(
      true,
      () => html`
        <b>ok</b>
      `)}
`;
```
To pure HTML, that would give this ugly thing
```html
<html>
  <span class="{"><span class="(">
      <span class=true></span>
      <span class="() => html">
        <span class="b ok b"></span>
      </span></span></span>
</html>
```

<img width="310" height="381" alt="Image" src="https://github.com/user-attachments/assets/bee293cf-ab96-424f-87a2-d6dbf2a0eebe" />

And this the requested indent of this HTML by `@html-eslint/indent`

https://html-eslint.org/playground#eyJjb2RlIjoiPGh0bWw+XG4gICAgPHNwYW4gY2xhc3M9XCJ7XCI+PHNwYW4gY2xhc3M9XCIoXCI+XG4gICAgICAgICAgICA8c3BhbiBjbGFzcz10cnVlPjwvc3Bhbj5cbiAgICAgICAgICAgIDxzcGFuIGNsYXNzPVwiKCkgPT4gaHRtbFwiPlxuICAgICAgICAgICAgICAgIDxzcGFuIGNsYXNzPVwiYiBvayBiXCI+PC9zcGFuPlxuICAgICAgICAgICAgPC9zcGFuPjwvc3Bhbj48L3NwYW4+XG5cbiAgICA8c3BhbiBjbGFzcz1cIntcIj48c3BhbiBjbGFzcz1cIihcIj5cbiAgICAgICAgICAgIDxzcGFuIGNsYXNzPXRydWU+PC9zcGFuPlxuICAgICAgICAgICAgPHNwYW4gY2xhc3M9XCIoKSA9PiBodG1sXCI+XG4gICAgICAgICAgICAgICAgPHNwYW4gY2xhc3M9XCJiIG9rIGJcIj48L3NwYW4+XG4gICAgPC9zcGFuPjwvc3Bhbj48L3NwYW4+XG48L2h0bWw+IiwiY29uZmlnIjp7InJ1bGVzIjp7IkBodG1sLWVzbGludC9pbmRlbnQiOiJlcnJvciJ9fSwibGFuZ3VhZ2UiOiJodG1sIn0=

This thing is not allowed
```html
<html>
  <span class="{"><span class="(">
      <span class=true></span>
      <span class="() => html">
        <span class="b ok b"></span>
  </span></span></span>
</html>
```

**Comment by yeonjuan** [comment]
@jpradelle 


I agree that the error in the ternary operator section, as shown below, was incorrect. I have corrected it and added it to the test cases. Thanks! (https://github.com/yeonjuan/html-eslint/pull/590/changes/34c22a3e80adf30520109356b4a61470b9dafac6)
<img width="312" height="185" alt="Image" src="https://github.com/user-attachments/assets/46c19f9e-1fd7-4874-9cd5-d5f27caf6b5f" />


>  closing brace only if this is the first item of the line. If there is some content before the closing brace, this is in the scope of the javascript linter to check the indent.

Hmm 😭. Indentation can be quite confusing depending on the perspective, but I find your proposed approach to be more consistent. 
Also, the current approach has the issue of requiring changes to string values even in places that are not HTML template literals.
let me fix it
<img width="258" height="194" alt="Image" src="https://github.com/user-attachments/assets/502dbf50-80dd-4b74-8b85-0cc3afa4e08d" />

**Comment by yeonjuan** [comment]
Hi  @jpradelle I fixed and it released in `0.60.0-alpha.2`
I would really appreciate it if you could double-check this for me.

I’ve verified this locally; here is a screenshot of the results

<img width="272" height="479" alt="Image" src="https://github.com/user-attachments/assets/73fad388-4272-49e3-b05f-7293194c5053" />

**Comment by jpradelle** [comment]
This looks great on a large project with version `0.60.0-alpha.2`, thanks a lot.

**Comment by jpradelle** [comment]
Oh no wait, I have an issue I think

**Comment by jpradelle** [comment]
In fact no issue, this work great, this revealed a bug in my code :)

**Comment by yeonjuan** [comment]
@jpradelle  Thank you for confirming. It will be included in the release.
